src/etwxlogging.c

```diff
@@ -14,6 +17,124 @@
 
 static LOGGER_LOG global_log_function = etwlogger_log;
 
+#ifdef LOGERROR_CAPTURES_STACK_TRACES
+#include "dbghelp.h"
+static volatile LONG doSymInit = 0; /*0 = not initialized, 1 = initializing, 2= initialized*/
+
+void* logging_malloc(size_t size)
+{
+    return malloc(size);
+}
+void logging_free(void* ptr)
+{
+    free(ptr);
+}
+
+#endif
+
+#if defined _MSC_VER
+#ifdef LOGERROR_CAPTURES_STACK_TRACES
+#define TRACE_MAX_STACK_FRAMES  1024
+#define TRACE_MAX_SYMBOL_SIZE   1024
+#define TRACE_MAX_STACK_AS_STRING_SIZE 10*1024
+#define TRACE_MAX_STACK_LINE_AS_STRING_SIZE 1024
+
+typedef union SYMBOL_INFO_EXTENDED_TAG
+{
+    SYMBOL_INFO symbol;
+    unsigned char extendsUnion[sizeof(SYMBOL_INFO) + TRACE_MAX_SYMBOL_SIZE - 1]; /*this field only exists to extend the size of the union to encompass "CHAR    Name[1];" of the SYMBOL_INFO to be as big as TRACE_MAX_SYMBOL_SIZE - 1 */ /*and the reason why it is not malloc'd is to exactly avoid a malloc that cannot be LogError'd (how does one log errors in a log function?!)*/
+}SYMBOL_INFO_EXTENDED;
+
+
+static SRWLOCK lockOverSymCalls = SRWLOCK_INIT;
+char* getStackAsString(void)
```

Comment:
>getStackAsString [](start = 6, length = 16)

I'd really like if this would not be duplicated. It really does the same thing like the one in xlogging, right?

So I suggest having a logging_stacktrace.c or similar where this code that gets the stack trace string built lives.

Reply:
Agree. Let me see about it.

---
In reply to: [276825068](https://github.com/Azure/azure-c-shared-utility/pull/302#discussion_r276825068) [](ancestors = 276825068)

Reply:
Seems like it is still duplicated.

---
In reply to: [276827635](https://github.com/Azure/azure-c-shared-utility/pull/302#discussion_r276827635) [](ancestors = 276827635,276825068)